turbopack/crates/turbo-tasks-fs/src/lib.rs

```diff
@@ -72,6 +72,81 @@
     rope::{Rope, RopeReader},
 };
 
+pub const MAX_FILE_NAME_LENGTH_UNIX: usize = 255;
+
+/// Validate the path, returning either the valid path, or a modified-but-now-valid path,
+/// or bailing with an error.
+///
+/// The behaviour of the file system changes depending on the OS, and indeed sometimes the FS
+/// implementation of the OS itself.
+///
+/// - On windows the limit for normal file paths is 260 characters, a holdover from the DOS days,
+///   but rust will opportunistically rewrite paths to 'UNC' paths for supported path operations
+///   which can be up to 32767 characters long.
+/// - On macOS, the limit is 255 characters traditionally, but after personal testing it seems to be
+///   around 298 on my (@arlyon) machine.
+/// - On linux, the limit differs between kernel (and by extension, distro) and filesystem. Individual files can be up to 255 bytes, while
+///   paths can be up to 4096 _on EXT4_. However, despite advertising these in `limits.h`, some filesystems actually support larger lengths.
+///   Fedora for example explicitly says not to program against the limit on linux (https://docs.fedoraproject.org/en-US/defensive-coding/tasks/Tasks-File_System/#sect-Defensive_Coding-Tasks-File_System-Limits)
```

Comment:
There's a lot of misinformation about this, but I believe `PATH_MAX` is just a meaningless historical artifact for some of the underlying POSIX APIs. EXT4 (and most other Linux FSes) can handle infinite length paths, as long as each segment is <255 bytes. This seems like the most accurate source: https://en.wikipedia.org/wiki/Comparison_of_file_systems#Limits